indexer.ts

```diff
@@ -0,0 +1,743 @@
+    let application = {
+      pot_id: receiverId,
+      applicant_id: data.project_id,
+      message: data.message,
+      submitted_at: data.submitted_at,
+      current_status: data.status,
+      tx_hash: receiptId,
+    };
+    const appl = await context.db.PotApplication.insert(application);
+
+    let activity = {
+      signer_id: signerId,
+      receiver_id: receiverId,
+      timestamp: application.submitted_at,
+      type: "Submit_Application",
+      action_result: appl[0].id.toString(), // result points to the pot application created.
+      tx_hash: receiptId, // should we have receipt on both action and activity?
+    };
+
+    await context.db.Activity.insert(activity);
+  }
+
+  async function handleApplicationStatusChange(
+    args,
+    receiverId,
+    signerId,
+    receiptId
+  ) {
+    let data = JSON.parse(args);
+    console.log("new factory data::", { ...data }, receiverId);
+
+    let receipt = block
+      .receipts()
+      .filter((receipt) => receipt.receiptId == receiptId)[0];
+    let update_data = JSON.parse(atob(receipt.status["SuccessValue"]));
+    let appl = (
+      await context.db.PotApplication.select({ applicant_id: data.project_id })
+    )[0];
+
+    let applicationReview = {
+      application_id: appl.id,
+      reviewer_id: signerId,
+      notes: update_data.notes,
+      status: update_data.status,
+      reviewed_at: update_data.updated_at,
+    };
+    let applicationUpdate = {
+      current_status: update_data.status,
+      last_updated_at: update_data.updated_at,
+    };
+
+    await context.db.PotApplicationReview.insert(applicationReview);
+    await context.db.PotApplication.update({ id: appl.id }, applicationUpdate);
+  }
+
+  async function handleDefaultListStatusChange(
+    args,
+    receiverId,
+    signerId,
+    receiptId
+  ) {
+    let data = JSON.parse(args);
+    console.log("update project data::", { ...data }, receiverId);
+
+    let listUpdate = {
+      default_registration_status: data.status,
+    };
+
+    await context.db.List.update({ id: receiverId }, listUpdate);
+  }
+
+  async function handleSettingPayout(args, receiverId, signerId, receiptId) {
+    let data = JSON.parse(args);
+    console.log("set payout data::", { ...data }, receiverId);
+    let payouts = data.payouts;
+    for (const payout in payouts) {
+      // general question: should we register projects as accounts?
+      let potPayout = {
+        recipient_id: payout["project_id"],
+        amount: payout["amount"],
+        ft_id: payout["ft_id"] || null,
+        tx_hash: receiptId,
+      };
+      await context.db.PotPayout.insert(potPayout);
+    }
+  }
+
+  async function handlePayout(args, receiverId, signerId, receiptId) {
+    let data = JSON.parse(args);
+    console.log("fulfill payout data::", { ...data }, receiverId);
+    let payout = {
+      recipient_id: data.project_id,
+      amount: data.amount,
+      paid_at: data.paid_at,
+      tx_hash: receiptId,
+    };
```

Comment:
i don't the args (`data`) will be in this format you're expecting here. the `transfer_payout_callback` method receives a `payout` argument so you may have to inspect this to get `project_id`, `amount` etc

Reply:
the `data` is the payout argument....

Reply:
but won't it be at `data.payout`?